crates/sequencer/src/block_template/block_assembly.rs

```diff
@@ -175,21 +175,27 @@ fn prepare_l1_segment(
     Ok(L1Segment::new(payloads))
 }
 
-fn load_header_record(h: u64, l1man: &L1BlockManager) -> Result<L1HeaderRecord, Error> {
+fn load_header_record(
+    blockid: &L1BlockId,
+    l1man: &L1BlockManager,
+) -> Result<L1HeaderRecord, Error> {
     let mf = l1man
-        .get_block_manifest(h)?
-        .ok_or(Error::MissingL1BlockHeight(h))?;
+        .get_block_manifest(blockid)?
+        .ok_or(Error::MissingL1Block(*blockid))?;
     // TODO need to include tx root proof we can verify
     Ok(L1HeaderRecord::create_from_serialized_header(
         mf.header().to_vec(),
         mf.txs_root(),
     ))
 }
 
-fn fetch_deposit_update_txs(h: u64, l1man: &L1BlockManager) -> Result<Vec<DepositUpdateTx>, Error> {
+fn fetch_deposit_update_txs(
+    blockid: &L1BlockId,
+    l1man: &L1BlockManager,
+) -> Result<Vec<DepositUpdateTx>, Error> {
```

Comment:
This function and all the surrounding logic would also be removed, since all this data is included in `ProtocolOperation`s.